### DISCUSSION TITLE
feature request: add a simple GUI to rclip

### DISCUSSION THREAD

**Discussion by 03302024** [yurijmikhalevich/rclip#253] [Ideas]
I am a Windows user. I installed the msi file. Now, how do I go about it?

**Comment by yurijmikhalevich** [comment]
Hi @03302024! Thank you for your interest in rclip ❤️ 

Since rclip is a command-line tool, you will have to use it from the terminal.

To use it:
- open any terminal (PowerShell, cmd)
- navigate to the directory you want to search using `cd`
- run `rclip <search query>`

This [YouTube video](https://www.youtube.com/watch?v=tAJHXOkHidw) shows it quite well. The idea is the same on Windows.

If you want to use the `rclip --preview` feature, you will have to install the [mintty](https://mintty.github.io/) terminal on Windows. I usually install it with the [Windows git distribution](https://git-scm.com/downloads/win). If you install it with git, it will be called "git bash."

Let me know if this helps 😄

**Comment by 03302024** [comment]
Thank you for your reply.

It seems the app downloads an additional ~1GB worth of files. Can I download them manually for future installations? can I backup the files that are downloaded now?

I installed [64-bit Git for Windows Setup](https://github.com/git-for-windows/git/releases/download/v2.47.0.windows.2/Git-2.47.0.2-64-bit.exe) with default settings.

I ran this command: "rclip -p face" and I get a list of image files, but no preview.

**Comment by 03302024** [comment]
This app really needs a gui. I found [rclip-server](https://github.com/ramayer/rclip-server), but there is no info on how to use it on Windows.

**Comment by yurijmikhalevich** [comment]
@03302024,

> I ran this command: "rclip -p face" and I get a list of image files, but no preview.

Did you run it in the "git bash" terminal?

> Can I download them manually for future installations? can I backup the files that are downloaded now?

Yes, you can back them up. The files are stored at `C:/Users/<USER>/AppData/Roaming/rclip`. There is also an SQLite db with rclip index; you probably wouldn't want to back up the index because it's specific per installation.

> This app really needs a gui.

I am considering building a GUI for it, but it will probably be a separate app. The whole point of `rclip` is to be used from a terminal. The benefit of this is that you can use it on any server/NAS over SSH.

> I found [rclip-server](https://github.com/ramayer/rclip-server), but there is no info on how to use it on Windows.

You will most likely need to install Python and setup `rclip-server` manually via `git clone git@github.com:ramayer/rclip-server.git`, installing dependencies, etc.

**Comment by 03302024** [comment]
> Did you run it in the "git bash" terminal?

Of course not! Now, it works just fine. Thank you.

> Yes, you can back them up.

You may want to add this info to the main web page.

> I am considering building a GUI for it

That would be great.

> but it will probably be a separate app.

Thats fine, as long as there is one download per app. I am not a fan of downloading after installating an app.

> You will most likely need to install Python and ...

As much as I like to do that, but I don't know anything of Linux. So, it is a no go for me.

**Comment by yurijmikhalevich** [comment]
@03302024,

> Of course not! Now, it works just fine. Thank you.

🎉 amazing! I am glad, it works!

> You may want to add this info to the main web page.

Good idea. May do. Or, I may bundle the model with the distribution. Will think about it 🤔 

I am trying to understand which kind of UI would work best for rclip. Can you help me by answering a few questions?

What are you using rclip for?

Which kind of UI would you prefer?

**Comment by 03302024** [comment]
> I may bundle the model with the distribution.

I prefer this. One complete download that works as intended after installation. One may adjust settings to his/her liking, then.

> What are you using rclip for?

Browsing my images and finding some specific ones.

> Which kind of UI would you prefer?

I think a simple practical, not a fancy, GUI would do. Something similar to rclip-server, however, I would add a button for selecting the "Search" folder, and another button for saving the image list as a text file. 

<img src="https://raw.githubusercontent.com/ramayer/rclip-server/refs/heads/main/doc/zebra_minus_stripes_plus_spots.png">

I would rather have a Treeview pane for navigation to different folders, and a Preview pane for showing a larger version of the image. Have an option to resize the thumbnails. Something more similar to XnViewMP.

<img src="https://www.xnview.com/img/screenshots/xnviewmp-win-01.jpg">

**Comment by 03302024** [comment]
Just for reference, two more screenshots from image comparison specialized apps.

![](https://camo.githubusercontent.com/bf5d54ab0b6434488f9d9952c95145a9a7c14eece1b40b7850e6e512896ce949/68747470733a2f2f65726d6967313937392e6769746875622e696f2f416e74694475706c2f646174612f68656c702f656e676c6973682f66696c65732f4d61696e466f726d2e706e67)

![](https://user-images.githubusercontent.com/46010672/129763067-8855a538-4a4f-4831-ac42-938eae9343bd.png)

**Comment by yurijmikhalevich** [comment]
@03302024, thank you for sharing your thoughts! I have noted everything and will let you know when / if the rclip GUI is released.

**Comment by 03302024** [comment]
Thank you. Looking forward to test a pre-release version.

**Comment by 03302024** [comment]
Is there any news on a Windows (portable) version with a GUI?

**Comment by yurijmikhalevich** [comment]
@03302024, thank you for reaching out! Not yet. I wonder if you would be open to a 20-minute user interview about rclip to help me better understand how you use and want to use it. Can you please email me at yurij@mikhalevi.ch with your timezone and a few good time slots if you are available?

**Comment by 03302024** [comment]
I am happy that you are still considering doing the GUI. I am very busy these days, however, we could continue our conversation here, as time permits. May be others also chime in, now with the new title.

Anyways, here are my quick thoughts, mostly taken from the previous discussions above:
- I like apps that are portable, i.e. one download that includes everything that is needed, and could be moved around to other computers, etc. Today, I tried to play with rclip again, but it started to download some extra (~1 GB) files again, so I just gave up. 
- I think a simple practical, not a fancy, GUI would do. Something similar to rclip-server, however, I would add a button for selecting the "Search" folder, and another button for saving the image list as a text file.
- I would rather have a Treeview pane for navigation to different folders, and also a Preview pane for showing a larger version of the image. Have an option to resize the thumbnails in the Results pane, and a button to toggle between thumbnail view and details view in the Results pane.

**Comment by yurijmikhalevich** [comment]
@03302024, I see your points! Thank you for sharing your thoughts 😊 

> I like apps that are portable, i.e. one download that includes everything that is needed, and could be moved around to other computers, etc

One thing is that because `rclip` can be updated much more frequently compared to the model itself, the `rclip` distribution doesn't include the model to make the updates smaller and faster. Your comment makes me think that providing an alternative distribution with the model included could be a good idea 🤔

**Comment by 03302024** [comment]
> providing an alternative distribution with the model included could be a good idea

Or, you could provide a direct link to the model files, in the app's release webpage. The model files could have version numbers and get updated as needed.

Then, you would also mention in the app's main webpage that one app file and one model file has to be downloaded by the user. As a portable app, one would put the extracted model files/folders in the app's folder.

**Comment by yurijmikhalevich** [comment]
> Then, you would also mention in the app's main webpage that one app file and one model file has to be downloaded by the user. As a portable app, one would put the extracted model files/folders in the app's folder.

@03302024, thank you for sharing your thoughts! _To me_ this sounds suboptimal; as a user I wouldn't want to have to make multiple downloads to start using the app. But it's helpful to know that this could be acceptable in some cases.

**Comment by 03302024** [comment]
> as a user I wouldn't want to have to make multiple downloads to start using the app

I totally agree, in general.

> But it's helpful to know that this could be acceptable in some cases.

In this specific case, there is an app which is relatively small in size and is frequently updated, and there is a model file that is huge in size and doesn't get updated as much.

Consider this scenario: a user intalls the app and the model file on one of his/her computers and loves it so much that wants to install it on his/her other computer as well. He/she will be very grateful not to have to download the huge model file again, rather just copy it from one computer to the other computer.

**Comment by yurijmikhalevich** [comment]
> a user intalls the app and the model file on one of his/her computers and loves it so much that wants to install it on his/her other computer as well. He/she will be very grateful not to have to download the huge model file again

This is fair! Thank you 😊

Can you please share what your internet speeds are? I'm trying to understand how much time it takes for you to download the model.

Also, I wonder if documenting the path where the model is saved in the README under the Advanced section and sharing instructions on how to copy it to a different machine would be a sufficient "quick fix" for this scenario in your opinion?

**Comment by 03302024** [comment]
>  Can you please share what your internet speeds are?

I have ADSL, and it is spotty at times.

> I wonder if documenting the path where the model is saved in the README under the Advanced section and sharing instructions on how to copy it to a different machine

That is a good idea, but having a zip file to download the model files/folders would have these benefits:

- The user would download at his/her own best time.
- The user may use a download manager for huge files, to be able to pause and resume the download process later.
- The user will know of the file sizes that are going to be downloaded, so no surprise there.
- A zip file may further (?) compress the files/folders to download.
- Rather than navigating the user through Windows folders, for a portable app, this quick note will suffice: "Put zip contents in the same directory as the app file."

**Comment by yurijmikhalevich** [comment]
ADSL sounds slow 😅 I am trying to understand how many users does this problem affect 🤔 

I'll give it more thought! Thank you for sharing your feedback 🙌

**Comment by 03302024** [comment]
> I am trying to understand how many users does this problem affect

Have a look at [Taking the Pulse of the Internet](https://www.maps.com/taking-the-pulse-of-the-internet/)

![Image](https://github.com/user-attachments/assets/5b13842d-7616-490e-b2c5-201e39045911)

**Comment by 03302024** [comment]
I would like to add this. For audio-to-text conversion, I am using this setup:

1- The app: [SubtitleEdit](https://github.com/SubtitleEdit/subtitleedit) or [PotPlayer](https://potplayer.tv/)
2- The engine: [whisper-standalone-win](https://github.com/Purfview/whisper-standalone-win)
3- The model: [Systran/faster-whisper-large-v2](https://huggingface.co/Systran/faster-whisper-large-v2)

You could also use something similar to that also.

**Comment by yurijmikhalevich** [comment]
@03302024, are you talking about an rclip UI or baking in the model within the distribution?

The UI can be out of scope for rclip OSS project at the current moment; I am still thinking what will make most sense; but you are welcome to build UI as a separate project!

As to the separate model download, can you add a link to the model to the README (under an advanced/hidden section) with the instructions on how to install it? Thank you!

**Comment by 03302024** [comment]
> are you talking about an rclip UI or baking in the model within the distribution?

What I am saying is I have figured out a setup for audio-to-text conversion, which works good for me. Now, I am suggesting a similar setup for rclip related stuff:

1- The app: A GUI to be developed like rclip-server that could work in Windows
2- The engine: rclip
3- The model: Models with separate download links. You could also have a fast/small model, an extended/large model, etc.

**Comment by yurijmikhalevich** [comment]
@03302024, thank you for sharing this. I will take some time to think about it.
